src/trio/_core/_run.py

```diff
@@ -1043,7 +1044,7 @@ async def __aexit__(
                 value.__context__ = old_context
                 # delete references from locals to avoid creating cycles
                 # see test_cancel_scope_exit_doesnt_create_cyclic_garbage
-                del _, combined_error_from_nursery, value, new_exc
+                del _, combined_error_from_nursery, value, new_exc, exc
```

Comment:
I think I misunderstood this. There's no cycle here because `exc.__traceback__` doesn't see this frame